.dockerignore

```diff
@@ -1,4 +1,31 @@
 vllm/*.so
 /.venv
-/build
-dist
+
```

Comment:
These additions should be safe and should make sure that `COPY . .` in the Dockerfiles does not copy any additional files that shouldn't go in a build.

I'm unsure on whether there's anything else we might need to add here to avoid copying temporary from the host into the container build (mostly thinking about `vllm/vllm_flash_attn` right now)